app/models/organization.rb

```diff
@@ -6,4 +6,17 @@ class Organization < ApplicationRecord
 
   has_many :chapters
   has_many :students
+
+  def add_user_with_role(email, role)
+    return false unless Role::LOCAL_ROLES.keys.include? role
+
+    user = User.find_or_create_by!(email: email)
+    return false if user.invalid? || user.member_of?(self)
```

Comment:
Is this `invalid?` check redundant because it would have been performed by `find_or_create_by!` above?

Reply:
Yeah, I think you're right.